opencog/moses/deme/deme_expander.cc

```diff
@@ -489,7 +489,7 @@ void deme_expander::optimize_demes(int max_evals, time_t max_time)
             // Optimize
             if (_params.atomspace_port) {
                 ComboToAtomeseConverter _to_atomese;
-                AtomSpace *as;
+                AtomSpace *as = new AtomSpace();
```

Comment:
Beware of the memory leaking here. You create an atomspace at each expansion but it is never deleted.

Reply:
But rather than constructing a new atomspace you should reuse the atomspace containing the populated features.

So I suppose at construction time of `deme_expander` (in `metapop_moses_results_b`) you'll have to somehow pass the atomspace used for populating the features and have `deme_expander` keep a pointer of it as attribute, then you'll be able to pass that atomspace pointer to the atomese converter, and optionally to `atomese_based_scorer`.

Reply:
currently the `populate-feature` is implemented for `table-problems` and reusing only the `atomspace` from `populate-features` is not enough I think especially considering we do have `demo-problems`. so, I was thinking about writing `common-setup` for `demo-problems` and creating atomspace there. For `table-problems` we can reuse the `populate-features` atomspace. Then changing the `metapop_moses_result_b` as you have said already.